Kover breaks Robolectric

Tested on Kover 0.4.4:
```
plugins {
    id 'org.jetbrains.kotlinx.kover' version '0.4.4'
}
```

Dependencies:
```
testImplementation "org.robolectric:robolectric:4.7.3"
```

Test
```
@RunWith(RobolectricTestRunner::class)
class RobolectricTest {
    @Test
    fun emptyTest() {
        "".toString()
    }
}
```

When I run this, I get this exception:
```
java.lang.IllegalAccessException: class org.robolectric.res.ResourceTableFactory cannot access a member of class android.R$styleable with modifiers "private static transient"
java.lang.RuntimeException: java.lang.IllegalAccessException: class org.robolectric.res.ResourceTableFactory cannot access a member of class android.R$styleable with modifiers "private static transient"
	at org.robolectric.res.ResourceTableFactory.addMissingStyleableAttributes(ResourceTableFactory.java:100)
	at org.robolectric.res.ResourceTableFactory.lambda$newFrameworkResourceTable$0(ResourceTableFactory.java:20)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.res.ResourceTableFactory.newFrameworkResourceTable(ResourceTableFactory.java:13)
	at org.robolectric.android.internal.AndroidTestEnvironment.getSystemResourceTable(AndroidTestEnvironment.java:394)
	at org.robolectric.android.internal.AndroidTestEnvironment.injectResourceStuffForLegacy(AndroidTestEnvironment.java:419)
	at org.robolectric.android.internal.AndroidTestEnvironment.loadAppPackage_measured(AndroidTestEnvironment.java:365)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$loadAppPackage$3(AndroidTestEnvironment.java:358)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.android.internal.AndroidTestEnvironment.loadAppPackage(AndroidTestEnvironment.java:358)
	at org.robolectric.android.internal.AndroidTestEnvironment.installAndCreateApplication(AndroidTestEnvironment.java:257)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$createApplicationSupplier$0(AndroidTestEnvironment.java:229)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$createApplicationSupplier$1(AndroidTestEnvironment.java:226)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:167)
	at org.robolectric.RuntimeEnvironment.getApplication(RuntimeEnvironment.java:71)
	at org.robolectric.android.internal.AndroidTestEnvironment.setUpApplicationState(AndroidTestEnvironment.java:194)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:325)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:265)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:88)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalAccessException: class org.robolectric.res.ResourceTableFactory cannot access a member of class android.R$styleable with modifiers "private static transient"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
	at java.base/java.lang.reflect.Field.checkAccess(Field.java:1075)
	at java.base/java.lang.reflect.Field.get(Field.java:416)
	at org.robolectric.res.ResourceTableFactory.addMissingStyleableAttributes(ResourceTableFactory.java:98)
```

When I run it without the kover plugin, it succeeds just fine.

--- a/README.md
+++ b/README.md
@@ -304,6 +304,7 @@ kover {
     jacocoEngineVersion.set("0.8.7")        // change version of JaCoCo agent and reporter
     generateReportOnCheck.set(true)         // false to do not execute `koverReport` task before `check` task
     disabledModules = setOf()               // setOf("module-name") to disable coverage for module with name `module-name`
+    instrumentAndroidPackage = false        // true to instrument packages `android.*` and `com.android.*`
 }
 ```
 </details>
@@ -319,6 +320,7 @@ kover {
     jacocoEngineVersion.set('0.8.7')        // change version of JaCoCo agent and reporter
     generateReportOnCheck.set(true)         // false to do not execute `koverReport` task before `check` task
     disabledModules = []                    // ["module-name"] to disable coverage for module with name `module-name`
+    instrumentAndroidPackage = false        // true to instrument packages `android.*` and `com.android.*`
 }
 ```
 </details>

--- a/src/main/kotlin/kotlinx/kover/KoverPlugin.kt
+++ b/src/main/kotlin/kotlinx/kover/KoverPlugin.kt
@@ -4,6 +4,7 @@
 
 package kotlinx.kover
 
+import kotlinx.kover.adapters.*
 import kotlinx.kover.api.*
 import kotlinx.kover.api.KoverPaths.HTML_AGG_REPORT_DEFAULT_PATH
 import kotlinx.kover.api.KoverNames.CHECK_TASK_NAME
@@ -264,7 +265,18 @@ class KoverPlugin : Plugin<Project> {
             else
                 null
         })
-        jvmArgumentProviders.add(CoverageArgumentProvider(this, agents, providers.koverExtension))
+
+        val excludeAndroidPackages =
+            project.provider { project.androidPluginIsApplied && !providers.koverExtension.get().instrumentAndroidPackage }
+
+        jvmArgumentProviders.add(
+            CoverageArgumentProvider(
+                this,
+                agents,
+                providers.koverExtension,
+                excludeAndroidPackages
+            )
+        )
 
         doLast(IntellijErrorLogChecker(taskExtension))
     }
@@ -293,8 +305,8 @@ private class IntellijErrorLogChecker(private val taskExtension: KoverTaskExtens
 private class CoverageArgumentProvider(
     private val task: Task,
     private val agents: Map<CoverageEngine, CoverageAgent>,
-    @get:Nested
-    val koverExtension: Provider<KoverExtension>
+    @get:Nested val koverExtension: Provider<KoverExtension>,
+    @get:Input val excludeAndroidPackage: Provider<Boolean>
 ) : CommandLineArgumentProvider, Named {
 
     @get:Nested
@@ -316,6 +328,19 @@ private class CoverageArgumentProvider(
             || koverExtensionValue.disabledModules.contains(task.project.name)
         ) {
             return mutableListOf()
+        }
+
+        if (excludeAndroidPackage.get()) {
+            /*
+            The instrumentation of android classes often causes errors when using third-party
+            frameworks (see https://github.com/Kotlin/kotlinx-kover/issues/89).
+
+            Because android classes are not part of the project, in any case they do not get into the report,
+            and they can be excluded from instrumentation.
+
+            FIXME Remove this code if the IntelliJ Agent stops changing project classes during instrumentation
+             */
+            taskExtensionValue.excludes = taskExtensionValue.excludes + "android.*" + "com.android.*"
         }
 
         return agents.getFor(koverExtensionValue.coverageEngine.get()).buildCommandLineArgs(task, taskExtensionValue)

--- a/src/main/kotlin/kotlinx/kover/adapters/PluginAdaptersFactory.kt
+++ b/src/main/kotlin/kotlinx/kover/adapters/PluginAdaptersFactory.kt
@@ -17,3 +17,8 @@ internal fun createAdapters(): List<CompilationPluginAdapter> {
         KotlinAndroidPluginAdapter()
     )
 }
+
+val Project.androidPluginIsApplied: Boolean
+    get() {
+        return plugins.findPlugin("android") != null || plugins.findPlugin("kotlin-android") != null
+    }

--- a/src/main/kotlin/kotlinx/kover/api/KoverExtension.kt
+++ b/src/main/kotlin/kotlinx/kover/api/KoverExtension.kt
@@ -45,6 +45,12 @@ open class KoverExtension(objects: ObjectFactory) {
      */
     @get:Input
     var disabledModules: Set<String> = emptySet()
+
+    /**
+     * Specifies whether the classes from 'android' and 'com.android' packages should be included if Android plugin is applied.
+     */
+    @get:Input
+    public var instrumentAndroidPackage: Boolean = false
 }
 
 public enum class CoverageEngine {

